processing/src/main/java/io/druid/segment/StringDimensionIndexer.java

```diff
@@ -534,79 +535,6 @@ public void fillBitmapsFromUnsortedEncodedArray(
     }
   }
 
-  @Override
-  public ValueMatcher makeIndexingValueMatcher(
-      final String matchValue,
-      final IncrementalIndexStorageAdapter.EntryHolder holder,
-      final int dimIndex
-  )
-  {
-    final int encodedVal = getEncodedValue(matchValue, false);
-    final boolean matchOnNull = Strings.isNullOrEmpty(matchValue);
-    if (encodedVal < 0 && !matchOnNull) {
-      return new BooleanValueMatcher(false);
-    }
-
-    return new ValueMatcher()
```

Comment:
Btw, you can cause situations like the one that leventov is describing in benchmarks by running the multiple code paths.  Once Java knows that it can be called multiple ways, it will undo whatever specializations it has done that no longer apply.  So, in the benchmark, if you actually benchmark things together in the same JVM you can see it.

Fwiw, when you find one of these, it can be fun to watch this happen and watch things speed up and slow down by adjusting the order of what runs first.  The first one run is sometimes always fastest, no matter which algorithm it is.